transport/messages/result_message_base.hh

```diff
@@ -41,6 +43,19 @@ public:
         _custom_payload[key] = value;
     }
 
+    void add_tablet_info(locator::tablet_replica_set tablet_replicas, std::pair<dht::token, dht::token> token_range) {
+        if (!tablet_replicas.empty() && token_range.first.data().size() == token_range.second.data().size() && token_range.first.data().size() == 8) {
```

Comment:
We shouldn't silently omit tablet info if sizes don't match because we'd lose shard-awarness silently and we won't notice it, or won't know why. 

I think we can throw an exception. Every request exercises this path, so we will learn that this doesn't work in the testing phase.

But see the next comment.